pkg/resourcemapping/operator_resource_mapping.go

```diff
@@ -266,20 +280,37 @@ func (ormClient *ORMClient) LocateOwnerPaths(ownedObj *unstructured.Unstructured
 	ownedKey := ownedObj.GetKind() + "/" + ownedObj.GetName() // source or owned resource
 
 	var allOwnerResourcePaths map[string][]devopsv1alpha1.ResourcePath
+	owners := map[string][]devopsv1alpha1.ResourcePath{}
 
-	orm, ownerObj, err := ormClient.retrieveORM(ownedObj, ownerReference)
+	orm, ownerObj, err := ormClient.retrieveOwnerResource(ownedObj, ownerReference)
 	if err != nil {
 		return allOwnerResourcePaths, err
 	}
 	if orm == nil {
-		return allOwnerResourcePaths, fmt.Errorf("Cannot find orm V1 for source: %s\n", ownedKey)
-	}
-	if orm == nil || ownerObj == nil {
-		return allOwnerResourcePaths, fmt.Errorf("Cannot find owner for source: %s\n", ownedKey)
+		return allOwnerResourcePaths, fmt.Errorf("cannot find orm V1 for source: %s", ownedKey)
```

Comment:
This method should consistently return either empty  resource paths structure or a default resource path with the source object, in case of any sorts of errors encountered while locating the path from the ORM.
In line 287, 290, it is returning empty set but a default source object further down in line 310.
Also add comments to the method describing the behavior of this method.

Reply:
Will address to return default source object to all conditions